### ISSUE TITLE
Config file writer

### ISSUE THREAD

**Issue by markfinger** [markfinger/python-webpack#14]
Need a way to deterministically write config files such that the path is constant so long as the content does not change.

``` python
from django_webpack.config_file import write_config_file

path = write_config_file(contents, prefix='some_identifier')

# `path` = '<BUNDLE_ROOT>/config_files/some_identifier_<hash_of_contents>.webpack.config.js'
```

markfinger/django-react#32

**Comment by markfinger** [comment]
**Might** need to nest the bundles up one dir, `BUNDLE_DIR = '<BUNDLE_ROOT>/bundles'`, so that potential collisions are avoided.

**Comment by markfinger** [comment]
At this stage, I'm inclined towards an API like

``` python
from webpack.config_file import ConfigFile, JS

config = Config_file(
    JS("var path = require('path');\n"),
    JS('module.exports = '), {
        context: JS("path.join('some', 'abs', 'path')"),
        entry: 'entry.js',
        output: {
            'path': '[bundle_dir]/react-components',
            'filename': '[name]-[hash].js',
        }, 
    }, JS(';\n')
)

config.render()  # Output the rendered content

config.write('/some/dir', prefix='foo', force=False)  # Write to a file, if it doesn't already exist
```

The above should generate

``` javascript
var path = require('path');
module.exports = {
    context: path.join('some', 'abs', 'path'),
    entry: 'entry.js',
    output: {
        'path': '[bundle_dir]/react-components',
        'filename': '[name]-[hash].js',
    }
};
```

Switching to raw JS which uses Node's path module would fix https://github.com/markfinger/python-react/issues/32 and https://github.com/markfinger/python-webpack/issues/18

**Comment by markfinger** [comment]
Fixed in 4.1.0
